docs/react/week-3/further-reading/container-components.md

```diff
@@ -0,0 +1,17 @@
+---
+id: container-components
+title: React Further Reading - Container Components
+---
+
+In real world applications, the things we want to remember in state follow the [_business logic_](https://en.wikipedia.org/wiki/Business_logic) required by our users. So for example the number of caught Pokemon in the exercise increases when you click on the button _Catch Pokemon_. Most of the time, business logic is about figuring out when and how to change state.
```

Comment:
This touches on one thing that I think we should emphasise more in the course:

React is all about breaking down problems into three things:
* Identifying state
* Deciding what to present given state
* Working out how/when/why to change state

I've seen many students able to follow instructions to create React components, but struggle to work out what components they should be writing or how they should be writing them, because they don't realise they should be trying to separate out these things.

In effect, our three weeks are each covering one of these (week 1 = what to present, week 2 = using state, week 3 = changing state). It feels like we can't _really_ teach this "how to design solutions" aspect until they've already seen the three separate pieces. Maybe this warrants turning React into a four-week class, where we spend week 4 talking about component design? I feel like this is too important to be a further reading footnote...

Reply:
This is a great suggestion, but as I think you're aware, we don't really have time to cover this in any depth.

> Maybe this warrants turning React into a four-week class, where we spend week 4 talking about component design?

This was an idea that came out of the London post-module retro too.

I would welcome proposals/PRs with more specifics! 😉 

---

Apologies, the rest of this comment isn't really React. Perhaps we should raise it on Slack/in another issue for wider visibility

I think structural changes like this would have to be discussed with the wider group. The 3 week constraint is more of a limit on volunteer time than anything else, so we'd have be careful with it (speaking from experience, I like to take some time off after the 3 weeks). I think the organisation in general would have concerns about further lengthening the course.

Additionally there's a couple of thoughts I have about it:

- I would feel a bit weird about having 4 weeks of React, but only 3 weeks of Node & DBs
- Not quite sure if putting it directly after the 3 existing weeks is the best choice
- Perhaps a "best practices" module at the end of the course might work better? This may also allow making it optional for students that are struggling?

Reply:
Definitely agree this is a discussion for another venue - wrote up https://codeyourfuture.slack.com/archives/C012UUW69S8/p1596893069021300 :)